File: LeetCode/Python/Easy/1422-Maximum-Score-After-Splitting-a-String/solution.py
class Solution:
    def maxScore(self, s: str) -> int:
        result=0

        n=len(s)

        zeros_on_the_left=[0 for _ in range(0, n)]

        zeros=0

        for (index, digit) in enumerate(s):
            if digit=='0':
                zeros+=1

            zeros_on_the_left[index]=zeros

        ones_on_the_right=[0 for _ in range(0, n)]

        ones=0

        for index in range(n-1, -1, -1):
            digit=s[index]
            
            if digit=='1':
                ones+=1

            ones_on_the_right[index]=ones
        
        for index in range(1, n):
            zeros=zeros_on_the_left[index - 1]

            ones=ones_on_the_right[index]

            total=zeros + ones

            result=max(result, total)

        return result

File: LeetCode/Rust/Easy/1422-Maximum-Score-After-Splitting-a-String/solution.rs
impl Solution
{
    pub fn max_score(s: String) -> i32
    {
        let mut result: i32=0;

        let n: usize=s.len();

        let mut zeros_to_the_left: Vec<u32>=vec![0; n];

        let mut zeros: u32=0;

        for (index, digit) in s.char_indices()
        {
            if digit=='0'
            {
                zeros+=1;
            }

            zeros_to_the_left[index]=zeros;
        }

        let mut ones_to_the_right: Vec<u32>=vec![0; n];

        let mut ones: u32=0;

        for (index, digit) in s.char_indices().rev()
        {
            if digit=='1'
            {
                ones+=1;
            }

            ones_to_the_right[index]=ones;
        }

        for index in 1..n
        {
            zeros=zeros_to_the_left[index - 1];

            ones=ones_to_the_right[index];

            let total: u32=ones + zeros;

            result=i32::max(result, total as i32);
        }

        result
    }
}